testApp/android/app/src/androidTest/java/com/example/reactnativedidomi/UIGettersTest.kt

```diff
@@ -36,51 +35,95 @@ class UIGettersTest {
     fun tearDown() {
     }
 
-    private fun testMethodCall(method: String, needToScroll: Boolean) {
-        onView(withText(method.toUpperCase())).perform(scrollTo(), click())
-
-        if (needToScroll)
-            onView(withText(method.toUpperCase())).perform(ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp())
-        waitForDisplayed(withText("$method-OK"))
-    }
-
     @Test
     fun test_GetDisabledPurposes() {
-        testMethodCall("getDisabledPurposes", true)
+        disagreeToAll()
+
+        tapButton("getDisabledPurposes".toUpperCase())
+        assertText(ALL_PURPOSE_IDS)
     }
 
     @Test
     fun test_GetDisabledPurposeIds() {
-        testMethodCall("getDisabledPurposeIds", true)
+        disagreeToAll()
+
+        tapButton("getDisabledPurposeIds".toUpperCase())
+        assertText(ALL_PURPOSE_IDS)
     }
 
     @Test
     fun test_GetDisabledVendors() {
-        testMethodCall("getDisabledVendors", true)
+        disagreeToAll()
+
+        tapButton("getDisabledVendors".toUpperCase())
+        assertText(ALL_VENDOR_IDS)
     }
 
     @Test
     fun test_GetDisabledVendorIds() {
-        testMethodCall("getDisabledVendorIds", true)
+        disagreeToAll()
+
+        tapButton("getDisabledVendorIds".toUpperCase())
+        assertText(ALL_VENDOR_IDS)
     }
 
     @Test
     fun test_GetEnabledPurposes() {
-        testMethodCall("getEnabledPurposes", true)
+        agreeToAll()
+
+        tapButton("getEnabledPurposes".toUpperCase())
+        assertText(ALL_PURPOSE_IDS)
     }
 
     @Test
     fun test_GetEnabledPurposeIds() {
-        testMethodCall("getEnabledPurposeIds", true)
+        agreeToAll()
+
+        tapButton("getEnabledPurposeIds".toUpperCase())
+        assertText(ALL_PURPOSE_IDS)
     }
 
     @Test
     fun test_GetEnabledVendors() {
-        testMethodCall("getEnabledVendors", true)
+        agreeToAll()
+
+        tapButton("getEnabledVendors".toUpperCase())
+        assertText(ALL_VENDOR_IDS)
     }
 
     @Test
     fun test_GetEnabledVendorIds() {
-        testMethodCall("getEnabledVendorIds", true)
+        agreeToAll()
+
+        tapButton("getEnabledVendorIds".toUpperCase())
+        assertText(ALL_VENDOR_IDS)
+    }
+
+    private fun agreeToAll() {
+        tapButton("setUserAgreeToAll".toUpperCase())
+        Thread.sleep(2000L)
```

Comment:
In Java / Kotlin you can insert underscore to make a number more readable.
2000L can be 2_000L
2000000 can be 2_000_000

Reply:
done